src/excmds.ts

```diff
@@ -3413,25 +3413,33 @@ function validateSetArgs(key: string, values: string[]) {
     const file = Metadata.everything.getFile("src/lib/config.ts")
     const default_config = file.getClass("default_config")
     const md = default_config.getMember(target[0])
+    function fallback(values) {
+        logger.warning("Could not fetch setting metadata. Falling back to type of current value.")
+        console.log("blah")
+        const currentValue = config.get(...target)
+        if (Array.isArray(currentValue)) {
+            return JSON.parse(values.join(" "))
+        } else if (currentValue === undefined || typeof currentValue === "string") {
+            return values.join(" ")
+        } else {
+            throw new Error("Unsupported setting type!")
+        }
+    }
     if (md !== undefined) {
         const strval = values.join(" ")
         // Note: the conversion will throw if strval can't be converted to the right type
-        if (md.type.kind === "object" && target.length > 1) {
+        if (md.type.kind === "object" && target.length == 2) {
```

Comment:
it feels obvious that this was the problem, but it clearly wasn't the only problem :(